CycloneDX/Runner.cs

```diff
@@ -511,5 +512,35 @@ internal static void AddMetadataTool(Bom bom)
             }
         }
 
+        /// <summary>
+        /// Updates all dependencies with version ranges to the version it was resolved to where possible.
+        /// </summary>
+        private static void ResolveDependencyVersionRanges(HashSet<DotnetDependency> runtimePackages)
```

Comment:
This will now resolve the packages and is basically the same code as before on the `ProjectAssetFileService`. Moving it ensures that it now really sees all packages, including build-only dev dependencies stripped by the `project.assets.json` generator.

It's important that this will still only resolve the packages for recursive project or solution mode. On non-recursive, these will still be missing and there is no way to fetch them. They COULD be added in non-recursive mode as `IsDevDependency=true` with `Scope=Exclude` but might still miss things like dependencies of the real package. So I (for now) decided against adding any workaround and only suggest the user in the warning to use `recursive` mode instead.